server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java

```diff
@@ -83,25 +80,23 @@
     public Stream<JmapResponse> process(JmapRequest request, ClientId clientId, MailboxSession mailboxSession) {
         Preconditions.checkArgument(request instanceof GetMailboxesRequest);
         GetMailboxesRequest mailboxesRequest = (GetMailboxesRequest) request;
-        TimeMetric timeMetric = metricFactory.timer(JMAP_PREFIX + METHOD_NAME.getName());
-        try (Closeable closeable =
-                 MDCBuilder.create()
-                     .addContext(MDCBuilder.ACTION, "GET_MAILBOXES")
-                     .addContext("accountId", mailboxesRequest.getAccountId())
-                     .addContext("mailboxIds", mailboxesRequest.getIds())
-                     .addContext("properties", mailboxesRequest.getProperties())
-                     .build()) {
-            return Stream.of(
-                    JmapResponse.builder().clientId(clientId)
-                    .response(getMailboxesResponse(mailboxesRequest, mailboxSession))
-                    .properties(mailboxesRequest.getProperties().map(this::ensureContainsId))
-                    .responseName(RESPONSE_NAME)
-                    .build());
-        } catch (IOException e) {
-            throw Throwables.propagate(e);
-        } finally {
-            timeMetric.stopAndPublish();
-        }
+        return metricFactory.withMetric(JMAP_PREFIX + METHOD_NAME.getName(),
+            () -> MDCBuilder.withMdc(
+                MDCBuilder.create()
+                    .addContext(MDCBuilder.ACTION, "GET_MAILBOXES")
+                    .addContext("accountId", mailboxesRequest.getAccountId())
+                    .addContext("mailboxIds", mailboxesRequest.getIds())
+                    .addContext("properties", mailboxesRequest.getProperties()),
+                () -> process(clientId, mailboxSession, mailboxesRequest)));
```

Comment:
closing a closable should be the responsability of the caller, no? Now it's not obvious that it is well closed.

Reply:
That's why it's only the MDCBuilder being passed ;-)

MDCBuilder is not closeable, thus managing the closeable is no responsibility of the caller code, but of the callee (to rephrase me, the MDCBuilder::withMDC method will generate the Closeable and close it, thus there is no responsibility mismatch).